columnflow/production/normalization.py

```diff
@@ -14,35 +18,112 @@
 maybe_import("scipy.sparse")
 ak = maybe_import("awkward")
 
+logger = law.logger.get_logger(__name__)
+
+
+def get_inclusive_dataset(dataset_inst: od.Dataset, stitching_datasets: list[od.Dataset]) -> od.Dataset:
+    """
+    Helper function to obtain the inclusive dataset from a list of datasets that are required to
+    stitch this *dataset_inst*.
+    """
+    process_map = {d.processes.get_first(): d for d in stitching_datasets}
+
+    process_inst = dataset_inst.processes.get_first()
+    incl_dataset = None
+    while process_inst:
+        if process_inst in process_map:
+            incl_dataset = process_map[process_inst]
+        process_inst = process_inst.parent_processes.get_first(default=None)
+
+    if not incl_dataset:
+        raise Exception("inclusive dataset not found")
+
+    unmatched_processes = {p for p in process_map if not incl_dataset.has_process(p, deep=True)}
+    if unmatched_processes:
+        raise Exception(f"processes {unmatched_processes} not found in inclusive dataset")
+
+    return incl_dataset
+
+
+def get_stitching_datasets(config_inst: od.Config, dataset_inst: od.Dataset) -> set[od.Dataset]:
+    """
+    Helper function to obtain all datasets that are required to stitch this *dataset_inst*.
+    """
+    stitching_datasets = {
+        d for d in config_inst.datasets
+        if (
+            d.has_process(dataset_inst.processes.get_first(), deep=True) or
+            dataset_inst.has_process(d.processes.get_first(), deep=True)
+        )
+    }
+    return stitching_datasets
+
+
+def get_br_from_inclusive_dataset(stats: dict) -> dict:
+    """
+    Helper function to compute the branching ratios from the inclusive sample.
+    """
+    # get the sum of weights inclusive and per process
+    sum_mc_weight = stats["sum_mc_weight"]
+    sum_mc_weight_per_process = stats["sum_mc_weight_per_process"]
+
+    if not sum_mc_weight == sum(sum_mc_weight_per_process.values()):
+        raise Exception(
+            "sum of event weights does not match the sum of event weights per process",
+        )
+
+    # compute the branching ratios
+    branching_ratios = {
+        int(process_id): sum_weights / sum_mc_weight
+        for process_id, sum_weights in sum_mc_weight_per_process.items()
+    }
+    # TODO: use keys of BRs to check if all processes are covered
```

Comment:
It would be good to have this check in place in this PR.

Reply:
I don't think you can check this at that point because it is unknown which types of process ids are expected to be used in the end, therefore you can only do this check during the event loop itself where you know which process ids are covered.